Chat_App/Scenes/Controller/ViewController.swift

```diff
@@ -0,0 +1,181 @@
+        bottomChatView.scrollToBottom()
+    }
+    
+    //MARK: - Functions
+    private func setUpLayout() {
+        topChatView.viewModel = viewModel
+        bottomChatView.viewModel = viewModel
+        switchButton.translatesAutoresizingMaskIntoConstraints = false
+        topChatView.translatesAutoresizingMaskIntoConstraints = false
+        bottomChatView.translatesAutoresizingMaskIntoConstraints = false
+        
+        view.addSubview(topChatView)
+        view.addSubview(centerLine)
+        view.addSubview(switchButton)
+        view.addSubview(bottomChatView)
+    }
+    
+    private func addDelegates() {
+        topChatView.sendMessageDelegate = self
+        bottomChatView.sendMessageDelegate = self
+        viewModel.delegate = self
+    }
+    
+    
+    //MARK: - Change background color
+    @objc private func switchButtonTapped() {
+        view.backgroundColor = switchButton.isOn ? Constants.Colors.darkMode : .white
+        
+        let chatViews = [topChatView, bottomChatView]
+        chatViews.forEach { chatView in
+            chatView.changeTextColor(color: switchButton.isOn ? Constants.Colors.darkModeTextColor : Constants.Colors.textColor)
+        }
+        
+        if switchButton.isOn {
+            userDefaults.set(true, forKey: "isOn")
+            
+        }else {
+            userDefaults.set(false, forKey: "isOn")
+        }
+    }
+
+    func sendButton(sender: UIButton) {
+        var message: Message
+        if sender === topTextView.getSendButton() && !topTextView.textView.text.isEmpty {
+            guard let text = topTextView.textView.text else { return }
+             message = Message(text: text, userId: 1, date: formattedDate, failedToSend: !NetworkManager.shared.isConnected)
+            viewModel.sendMessages(with: message)
+            topTextView.textView.text = ""
+        } else if sender === bottomTextView.getSendButton() && !bottomTextView.textView.text.isEmpty {
+            guard let text = bottomTextView.textView.text else { return }
+             message = Message(text: text, userId: 2, date: formattedDate, failedToSend: !NetworkManager.shared.isConnected)
+            viewModel.sendMessages(with: message)
+            bottomTextView.textView.text = ""
+        }
+    }
+    
+  
+    func hideKeyboard() {
+          let tap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
+          tap.cancelsTouchesInView = false
+          view.addGestureRecognizer(tap)
+      }
+
+      @objc func dismissKeyboard() {
+          view.endEditing(true)
+      }
+    
+    func checkSwithcState() {
+        if userDefaults.bool(forKey: "isOn") {
+            switchButton.isOn = false
+            view.backgroundColor = Constants.Colors.darkMode
+        } else {
+            switchButton.isOn = true
+            view.backgroundColor = .white
+        }
+    }
+    
+    //MARK: - Set up constraints
+    private func constraintsAssigner() {
```

Comment:
naming issue there: `constraintsAssigner` is an entity name. You should name methods by verbs and describe what they do